corehq/apps/api/tests/user_resources.py

```diff
@@ -691,3 +691,30 @@ def test_domain_not_returned_when_flag_not_enabled(self, *args):
         bundle.request.user = self.user
         resp = UserDomainsResource().obj_get_list(bundle)
         self.assertEqual(0, len(resp))
+
+
+class TestCommCareAnalyticsUserRolesResource(APIResourceTest):
+    resource = v0_5.CommCareAnalyticsUserRolesResource
+    api_name = 'v0.5'
+
+    def test_flag_not_enabled(self):
+        self.assertTrue(self.user.domain_memberships[0].is_admin)
+
+        response = self._assert_auth_get_resource(self.list_endpoint)
+        self.assertEqual(response.status_code, 403)
+
+    @flag_enabled('SUPERSET_ANALYTICS')
+    def test_user_roles_returned(self):
+        self.assertTrue(self.user.domain_memberships[0].is_admin)
```

Comment:
nit: I believe we would also want to test this for a non admin?

Reply:
@mkangia Is that necessary, since the `get_commcare_analytics_access_for_user_domain` function is covered by the relevant tests, and that is the only determining factor for what roles/permissions is returned?

Reply:
oh cool :+1: